### ISSUE TITLE
Using with webpack crashes a page

### ISSUE THREAD

**Issue by mazavr** [sciactive/pnotify#212]
I'm using webpack with pnotify (I get pnotify 3.0.0 from npm registry) in my project. So, when I use pnotify 3.0.0 it crashes a site.
The reason is here (pnotify.js line 865)

```
if (root.document.body) {
    do_when_ready();
} else {
    $(do_when_ready);
}
```

because root is only an `object` in my case...
If I fix it by adding 

```
if (root.document && root.document.body) {
```

if crashes again later on (jwindow.height() call). I guess 

```
jwindow = $(root);
```

on 45 line is the reason (where root is only an `object`)

**Comment by xtr0** [comment]
@mazavr 
To workaround this issue you can do the following:

code:
```
var PNotify = require("pnotify/src/pnotify.js");

// load pnotify modules (I'm using only this two)
require("pnotify/src/pnotify.mobile.js");
require("pnotify/src/pnotify.buttons.js");
// ... other used pnotify modules if any ...

```
webpack config:

```
...
    module   : {
        loaders: [
            { test: /pnotify\/src\/.+\.js$/ , loader: "imports?define=>false,global=>window" },
...
       ]
   }
```

It works for me

**Comment by tasarsu** [comment]
+1
Hi, same issue here with webpack got Uncaught TypeError: Cannot read property 'body' of undefined. @xtr0 I tried your way but didnt worked for me

**Comment by tasarsu** [comment]
@xtr0 I guess in pnotify mobile module `require('./pnotify')` is called which caused error in my code. I managed to solve this issue by changing the regex. Thank you for the clue

``` javascript 
var PNotify = require("pnotify/src/pnotify.js");
require("pnotify/src/pnotify.mobile.js");
require("pnotify/src/pnotify.buttons.js");
require("pnotify/src/pnotify.desktop.js");
```


``` javascript 

    loaders: [
      { 
        test: /pnotify.*\.js$/, 
        loader: "imports?define=>false,global=>window" 
      },
```

**Comment by ggrimbert** [comment]
Hi,

I am having the same problem, but when I tried your way, but I have the following : 
```
Unhandled rejection Error: Cannot find module "pnotify/src/pnotify.js"
```

In my bundle.js I have the following : 
```
var PNotify = __webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"pnotify/src/pnotify.js\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()));
	__webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"pnotify/src/pnotify.mobile.js\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()));
	__webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"pnotify/src/pnotify.buttons.js\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()));
	__webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"pnotify/src/pnotify.desktop.js\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()));
```

I tried setting the complete path to the files, but nothing works...

**Comment by istrau3** [comment]
+1 same issue. would love to avoid hacking around this.

**Comment by istrau3** [comment]
I am using the solution above but cannot seem to make the desktop or the animations work. Would love to get this resolve. I am using this in an Aurelia project which relies on webpack for all the development workflow.

**Comment by starplanet** [comment]
+1

**Comment by bfgh2678** [comment]
+1

**Comment by YoeriVD** [comment]
+1

**Comment by Pirozek** [comment]
I don't think that this will ever get solved, PNotify looks like its no longer maintained by its author. Any suggestion for better library?

**Comment by AliveCorpse** [comment]
This works for me, but i had installed 'pnotify' by npm:

 `require('expose?PNotify!imports?define=>false,global=>window,$=jquery!pnotify');`

**Comment by Pirozek** [comment]
I finally made it working by adding this loader:
```javascript
test: /pnotify\.js/,
loader: 'imports?global=>window,this=>window'
```

And then adding jQuery and PNotify to Provide plugin:
```
new ProvidePlugin({
			jQuery: 'jquery',
			$: 'jquery',
			jquery: 'jquery',
			"window.jQuery": "jquery",
			PNotify: 'pnotify'
		})
```

I was trying to make PNotify work with angular-pnotify library, which expects PNotify to be on global scope (for `new PNotify()`). Hope this helps to somebody who come to the same problem after me :)

**Comment by nathan818fr** [comment]
For those who have the problem, I proposed a fix here: https://github.com/sciactive/pnotify/pull/268

**Comment by alfeg** [comment]
@nathan818fr , @hperrin Can someone update npm package please? It's seems to not have this fix =(

**Comment by sman591** [comment]
Until a new npm version is released, I got it working with this:

`package.json`
```json
{
  "devDependencies": {
    "pnotify": "git+https://git@github.com:sciactive/pnotify.git#175af26286f584f33282a686191715fadae2f67b"
  }
}
```

`webpack.config.js`
```javascript
module.exports = {
...
  plugins: [
    new webpack.ProvidePlugin({
      $: "jquery",
      jQuery: "jquery"
    })
  ]
...
```

`index.js`
```javascript
import $ from 'jquery';
import PNotify from 'pnotify';
window.$ = $;
window.jQuery = $;
window.PNotify = PNotify;
```

**Comment by codymullins** [comment]
For anyone who has trouble with the link @sman591 suggested I had to use the following instead:
`git+https://github.com/sciactive/pnotify.git#175af26286f584f33282a686191715fadae2f67b`

**Comment by joelcuevas** [comment]
Updating to 3.2.0 fixed this for me. 👍

**Comment by hperrin** [comment]
Yes, should be fixed in the new release.

**Comment by ryancwalsh** [comment]
I'm using Webpack (in Laravel Mix) and can't seem to get PNotify working.  Should I be using something other than `var PNotify = require('pnotify');` after I've run `npm i pnotify`? Thanks.

**Comment by ryancwalsh** [comment]
Oops, I made a stupid mistake. I was trying to copy-and-paste PNotify code from somewhere else, and it must have been code that only worked with a previous version of PNotify.

Now I've got it working by importing this in my SCSS: `@import "../../../../node_modules/pnotify/dist/PNotifyBrightTheme.css";`

And in my JS I use `var PNotify = require('pnotify/dist/umd/PNotify');` and `PNotify.alert('Notice me, senpai!');`.
